### PR TITLE
Make it compatible with @ember/test-helpers and ember-qunit v5

### PR DESCRIPTION
**Pull request by SergeAstapov**
Unblocks san650/ember-cli-page-object#524

### CODE REVIEW

**Comment by SergeAstapov** [comment]
Because ember-qunit v5 has complex setup, this addon is not used by modern apps it does not seem feasible to spend time at this point and add ember-try scenario.

@cibernox would you be able to release a patch version of this so that apps using `ember-cli-page-object` could upgrade to Ember 3.24?

File: addon-test-support/-private/compatibility.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+// This is a wrapper around `@ember/test-helpers` that we need for compatibility
+// reasons. Apps and addons using ember-qunit v4 aren't supposed to depend directly on
+// `@ember/test-helpers` and just use the one that their version of
+// `ember-qunit` or `ember-mocha` provides.
+// Apps and addons using ember-qunit v5 directly depend on
+// `@ember/test-helpers`.
+const { require } = window;
```
Comment (rwjblue) [inline addon-test-support/-private/compatibility.js:8]: Usually you'd `import require from 'require';` for this

**Review by rwjblue — COMMENTED** [review]
Have you tried migrating from wait to settled (from `@ember/test-helpers`) instead? I think that should work for both majors (though it probably didn't work at the time of originally authoring).

**Comment by SergeAstapov** [comment]
@rwjblue safari as I remember `settled` has same behavior as `wait` so should be simple swap.
BTW would it require major version bump? I'm not sure if it's small enough change

File: addon-test-support/-private/compatibility.js (hunk 1 above)
Comment (SergeAstapov) [reply addon-test-support/-private/compatibility.js]: @rwjblue ah yes, thank you! This is fixed

File: addon-test-support/-private/compatibility.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+if (require.has('ember-test-helpers/wait')) {
+  // This is implemented as a function that calls `ember-test-helpers/wait`
+  // rather than just assigning `helpers.wait = require(...).default` because
+  // since this code executes while modules are initially loading, under certain
+  // conditions `ember-test-helpers/wait` can still be in the pending state
+  // at this point, so its exports are still undefined.
+  waitFn = (...args) => require('ember-test-helpers/wait').default(...args);
+} else if (require.has('@ember-test/helpers')) {
+  waitFn = (...args) => require('@ember-test/helpers').wait(...args);
```
Comment (ro0gr) [inline addon-test-support/-private/compatibility.js:19]: must be `@ember/test-helpers`? and a line above.

Comment (SergeAstapov) [reply addon-test-support/-private/compatibility.js]: @ro0gr 🤦 thanks for noticing! this is fixed now

**Comment by cibernox** [comment]
Merged and released in 0.7.0

**Pull request merged**